File: 1.0/src/c/cz/mg/vulkan/PFN_vkUpdateDescriptorSets.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkUpdateDescriptorSets_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jint descriptorWriteCount, jlong pDescriptorWrites, jint descriptorCopyCount, jlong pDescriptorCopies){
    (void)env;
    (void)jc;
    PFN_vkUpdateDescriptorSets f = (PFN_vkUpdateDescriptorSets)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        (uint32_t)descriptorWriteCount,
        ((VkWriteDescriptorSet*)jniLongToPointer(pDescriptorWrites)),
        (uint32_t)descriptorCopyCount,
        ((VkCopyDescriptorSet*)jniLongToPointer(pDescriptorCopies))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkUpdateDescriptorSets.java
package cz.mg.vulkan;

public class PFNvkUpdateDescriptorSets extends VkFunctionPointer {
    public PFNvkUpdateDescriptorSets() {
    }

    protected PFNvkUpdateDescriptorSets(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkUpdateDescriptorSets(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkUpdateDescriptorSets(long value) {
        setValue(value);
    }

    public PFNvkUpdateDescriptorSets(VkInstance instance) {
        super(instance, new VkString("vkUpdateDescriptorSets"));
    }

    public void call(VkDevice device, int descriptorWriteCount, VkWriteDescriptorSet pDescriptorWrites, int descriptorCopyCount, VkCopyDescriptorSet pDescriptorCopies){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), descriptorWriteCount, pDescriptorWrites != null ? pDescriptorWrites.getVkAddress() : VkPointer.NULL, descriptorCopyCount, pDescriptorCopies != null ? pDescriptorCopies.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long device, int descriptorWriteCount, long pDescriptorWrites, int descriptorCopyCount, long pDescriptorCopies);

}
